File: 53.最大子数组和.py
#
# @lc app=leetcode.cn id=53 lang=python3
#
# [53] 最大子数组和
#
from typing import List

# @lc code=start
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        res = nums[0]
        pre = 0
        for v in nums:
            pre = max(pre + v, v)
            res = max(res, pre)
        return res
# @lc code=end

File: 53.最大子数组和.rs
/*
 * @lc app=leetcode.cn id=53 lang=rust
 *
 * [53] 最大子数组和
 */

// @lc code=start
impl Solution {
    pub fn max_sub_array(nums: Vec<i32>) -> i32 {
        let mut pre = 0;
        let mut res = nums[0];
        for v in nums {
            pre = max(pre + v, v);
            res = max(res, pre);
        }
        res
    }
}
// @lc code=end

